rcl/src/rcl/wait.c

```diff
@@ -296,6 +301,22 @@ rcl_wait_set_get_allocator(const rcl_wait_set_t * wait_set, rcl_allocator_t * al
   } \
   memset(wait_set->impl->RMWStorage, 0, sizeof(void *) * Type ## s_size);
 
+#define CHECK_DOUBLE_USAGE(Type) \
```

Comment:
minor: add a comment explaining how this check works... it took me a couple of reads to understand it.
something like
```
Ensure that the waitset lists do not contain duplicated entries.
For example, considering `wait_set->clients`:
 - first we set `in_use_by_waitset` to false for each entry.
 - then we loop again setting `in_use_by_waitset` to true one by one
 - if we find an entry where `in_use_by_waitset` was already true, it means that it was present twice in the waitset and we return an error

```

Reply:
added